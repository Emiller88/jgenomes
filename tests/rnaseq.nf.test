nextflow_pipeline {

    name "Build rnaseq references"
    script "../main.nf"
    tag "pipeline"

    test("Run with profile test for rnaseq | --input assets/genomes/test/pipelines/default.yml") {

        when {
            params {
                input  = "${projectDir}/assets/genomes/test/pipelines/default.yml"
                outdir = "$outputDir"
                tools  = 'bbmap,bowtie1,bowtie2,faidx,gffread,hisat2,hisat2_extractsplicesites,kallisto,make_transcripts_fasta,rsem,salmon,sizes,star'
                // TODO REMOVE ME when the module is fixed
                // cf https://github.com/nf-core/modules/issues/7045
                hisat2_build_memory = "3.GB"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
    test("Run with profile test for rnaseq | --input assets/genomes/test/pipelines/default_no_splice_sites.yml") {

        when {
            params {
                input  = "${projectDir}/assets/genomes/test/pipelines/default_no_splice_sites.yml"
                outdir = "$outputDir"
                tools  = 'hisat2,hisat2_extractsplicesites'
                // TODO REMOVE ME when the module is fixed
                // cf https://github.com/nf-core/modules/issues/7045
                hisat2_build_memory = "3.GB"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
