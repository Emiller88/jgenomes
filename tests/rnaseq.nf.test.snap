{
    "Run with profile test for rnaseq | --input GRCh38_chr21.yml": {
        "content": [
            6,
            {
                "BOWTIE1_BUILD": {
                    "bowtie": "1.3.1"
                },
                "BOWTIE2_BUILD": {
                    "bowtie2": "2.5.2"
                },
                "GFFREAD": {
                    "gffread": "0.12.7"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "STAR_GENOMEGENERATE": {
                    "star": "2.7.11b",
                    "samtools": 1.2,
                    "gawk": "5.1.0"
                },
                "Workflow": {
                    "nf-core/references": "v1.0dev"
                }
            },
            [
                "bowtie1",
                "bowtie1/bowtie",
                "bowtie1/bowtie/GRCh38_chr21.1.ebwt",
                "bowtie1/bowtie/GRCh38_chr21.2.ebwt",
                "bowtie1/bowtie/GRCh38_chr21.3.ebwt",
                "bowtie1/bowtie/GRCh38_chr21.4.ebwt",
                "bowtie1/bowtie/GRCh38_chr21.rev.1.ebwt",
                "bowtie1/bowtie/GRCh38_chr21.rev.2.ebwt",
                "bowtie2",
                "bowtie2/bowtie2",
                "bowtie2/bowtie2/GRCh38_chr21.1.bt2",
                "bowtie2/bowtie2/GRCh38_chr21.2.bt2",
                "bowtie2/bowtie2/GRCh38_chr21.3.bt2",
                "bowtie2/bowtie2/GRCh38_chr21.4.bt2",
                "bowtie2/bowtie2/GRCh38_chr21.rev.1.bt2",
                "bowtie2/bowtie2/GRCh38_chr21.rev.2.bt2",
                "fasta_fai",
                "fasta_fai/GRCh38_chr21.fa.fai",
                "fasta_sizes",
                "fasta_sizes/GRCh38_chr21.fa.sizes",
                "gffread",
                "gffread/GRCh38_chr21.gtf",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml",
                "star",
                "star/star",
                "star/star/Genome",
                "star/star/Log.out",
                "star/star/SA",
                "star/star/SAindex",
                "star/star/chrLength.txt",
                "star/star/chrName.txt",
                "star/star/chrNameLength.txt",
                "star/star/chrStart.txt",
                "star/star/exonGeTrInfo.tab",
                "star/star/exonInfo.tab",
                "star/star/geneInfo.tab",
                "star/star/genomeParameters.txt",
                "star/star/sjdbInfo.txt",
                "star/star/sjdbList.fromGTF.out.tab",
                "star/star/sjdbList.out.tab",
                "star/star/transcriptInfo.tab"
            ],
            [
                "GRCh38_chr21.1.ebwt:md5,32b67a141c03f8f97d6f3fc4c9b90796",
                "GRCh38_chr21.2.ebwt:md5,d811357a63a9429006f2f7a84255ef91",
                "GRCh38_chr21.3.ebwt:md5,dbf5af96efd98d6b03f1e5d2baed848a",
                "GRCh38_chr21.4.ebwt:md5,bd952f748321c4c230671e8fce7b1650",
                "GRCh38_chr21.rev.1.ebwt:md5,5e5a45602d63d358e3edca7cdb683a1b",
                "GRCh38_chr21.rev.2.ebwt:md5,1e5088944137b3277e65306ea1a347a5",
                "GRCh38_chr21.1.bt2:md5,3de5dca5da6810e9addac0df4c682d05",
                "GRCh38_chr21.2.bt2:md5,c61d725cb17a44cf928f89806fe37997",
                "GRCh38_chr21.3.bt2:md5,dbf5af96efd98d6b03f1e5d2baed848a",
                "GRCh38_chr21.4.bt2:md5,bd952f748321c4c230671e8fce7b1650",
                "GRCh38_chr21.rev.1.bt2:md5,652b3b94689d5ca59b02a7f89e799958",
                "GRCh38_chr21.rev.2.bt2:md5,33a970574d59232fdfab1bf2cb26d7ae",
                "GRCh38_chr21.fa.fai:md5,6ea51e1c1231da5e49212ea9f493d703",
                "GRCh38_chr21.fa.sizes:md5,c985a141685e8431ec27c782816cb744",
                "GRCh38_chr21.gtf:md5,c552c0aee5f1becb1d3991d21e63a5c1",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "Genome:md5,612664e3cfde5e1b73ad541d93752b31",
                "SA:md5,074ae54177bb7b9cb981382f043f36e5",
                "SAindex:md5,1c9cd646313f1abb1cfc205ccab73464",
                "chrLength.txt:md5,b0be0a56ddefa84552742c72d4859eac",
                "chrName.txt:md5,e99d7d1051eee43ceab5563c2d09fcee",
                "chrNameLength.txt:md5,c985a141685e8431ec27c782816cb744",
                "chrStart.txt:md5,6925b594ea2eeb964ba87cd6d42ab98f",
                "exonGeTrInfo.tab:md5,a36b92eeceaaf921b9d19cd8e806a98e",
                "exonInfo.tab:md5,bb892190d1ebdd59bf55652916121479",
                "geneInfo.tab:md5,4303e97d841035b5c0fcc3686b2f5a36",
                "genomeParameters.txt:md5,79725c6405e5c67a8cc75c0ee45f73c2",
                "sjdbInfo.txt:md5,12fb05dc7cea89735a0c19e1c0df61cb",
                "sjdbList.fromGTF.out.tab:md5,d9e4a184cde15a5ab282ff66e740a4a2",
                "sjdbList.out.tab:md5,766fbca932681f8666b3a9e5fb3640bd",
                "transcriptInfo.tab:md5,ad9baa68c5432908b42693edb9aed02a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-11-28T13:47:40.681771583"
    }
}